lineapy/graph_reader/artifact_collection.py

```diff
@@ -46,121 +39,141 @@ class ArtifactCollection:
     def __init__(
         self,
         db: RelationalLineaDB,
-        target_artifacts: Union[
-            List[str], List[Tuple[str, int]], List[Union[str, Tuple[str, int]]]
-        ],
+        target_artifacts: Sequence[Union[str, Tuple[str, int]]],
```

Comment:
Since this ticket is for refactoring, do you think we should just use `ineapy.api.models.linea_artifact.LineaArtifactDef` instead of `Union[str, Tuple[str, int]]`

Reply:
Yes, I can move the LineaArtifactDef search to outside the Artifact Collection so that all our internal classes use the ArtifactDef. Makes the type hinting much cleaner and is something we should conform too as a standard.